### DISCUSSION TITLE
Remote Ollama server: Model Selection Issue

### DISCUSSION THREAD

**Discussion by SamuelDevdas** [open-webui/open-webui#2359] [General]
# Bug Report

## Description

**Bug Summary:**
[When using ollama installed on remote server using docker, how to select models?(
![Screenshot 2024-05-17 225548](https://github.com/open-webui/open-webui/assets/84823680/bee13746-d26f-42a8-be68-ccdb82b9c46b)
)]
![WhatsApp Image 2024-05-17 at 23 00 55_99cf826f](https://github.com/open-webui/open-webui/assets/84823680/fb0068aa-c518-4d7d-a351-c5d2762ff7b2)

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

**Expected Behavior:**
[Available models should be listed, or an options while setting up docker container should be available]

**Actual Behavior:**
[Models not available to select, thus chat cant be started]

## Environment

- **Open WebUI Version:** [e.g., 0.1.120]
- **Ollama (if applicable):** [e.g., 0.1.30, 0.1.32-rc1]

- **Operating System:** [raspbian]
- **Browser (if applicable):** [e.g., Chrome 100.0]

## Reproduction Details

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by JackXie-personal** [comment]
The same problem, I don't know how to solve it

**Reply by SamuelDevdas** [reply]
Thanks! Please upvote the discussion and the issue created.

**Comment by yz1797203443** [comment]
![image](https://github.com/open-webui/open-webui/assets/73625338/d005abd1-01ee-4e9b-a81e-6def3c2fa2b2)
backend.apps.ollama.main.get_ollama_tags

**Reply by yz1797203443** [reply]
Can you take a look at the method of returning the model

**Comment by JackXie-personal** [comment]
I may find the reason.
When ollama is installed in the docker, the ollama base url should be 'http://host.docker.internal:11434'
Hope it can help you.

**Reply by MonkeyMatrix** [reply]
Sometimes you may even have problems with this, so you may also try http://127.0.0.1:11434. Depends on how everything is set up though. You could try both solutions.

**Reply by SamuelDevdas** [reply]
Thanks, but as you would notice (see picture: "NO RESULTS FOUND) I have the ui running locally using remote ollama server,  but can't access the models in the dropdown. The ACTUAL issue is not able to detect and select the model list.

**Reply by MonkeyMatrix** [reply]
"No results found" may refer to it not finding the instance in the first place. Have you attempted to use host.docker.internal and 127.0.0.1/localhost? It seems there is a "refresh" button beside the base url in the connections tab. This appears to try to connect to the instance in the input box. Could you try both "http://localhost:11434" and "http://host.docker.internal:11434" and press the "refresh" button after using each? It should let you know if it was successful or not. However, in the case the instance is running on a different machine you may need to use something else.

Edit: You may also need --network=host in this case though. Sorry I am not more experienced with Docker!
